File: 1310/1310.py

def max_subsum(lista):
    if min(lista)>=0:
        return sum(lista)

    soma_maxima = 0
    soma_atual = 0
    for i in range(0,len(lista)-1):
        if soma_atual> soma_maxima:
            soma_maxima = soma_atual
        if lista[i] + soma_atual<0:
            soma_atual = 0
        else:
            soma_atual = soma_atual + lista[i]

    return soma_maxima


def main():
    while(True):
        try:
            n = int(input())
        except EOFError: 
            break
        custo = int(input())
        diarias = []
        for i in range(0,n):
            diarias.append(int(input())-custo)
        resposta = max_subsum(diarias)
        print(resposta)

if __name__ == "__main__":
    main()

File: 1310/1310.cpp
#include <iostream>
using namespace std;

int max_subsum(int *lista,int n){
   int maxima_soma = 0;
   int soma_atual = 0;

    for (int i = 0; i<n; i++){
        if(lista[i]+soma_atual<0){
            soma_atual = 0;
        }else if(lista[i]+soma_atual>maxima_soma){
            maxima_soma = lista[i]+soma_atual;
            soma_atual = lista[i]+soma_atual;
        }else{
            soma_atual = lista[i]+soma_atual;
        }      
    }
    return maxima_soma;
}
int main(){
    int n;
    while(cin>>n){
        int custo,receita[n+1];
        cin>>custo;
        for (int i = 0; i<n ; i++){
            int aux;
            cin>>aux;
            receita[i] = aux - custo;
        }
        cout<<max_subsum(receita,n)<<endl;
    }
}